Added new test to verify SRIOV failsafe for DPDK

rmsivakanth opened the pull request.

rmsivakanth commented:
No errors found in PSScriptAnalyzer tool run.

rmsivakanth commented:
Execution is verified.
```
09/20/2018 05:26:17 : [INFO ] CleanUP Successful for ICA-RG-DPDK-SRU18-KERA-636729899893..
09/20/2018 05:26:17 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-SRIOV-FAILSAFE-FOR-DPDK~~~~~~~~~~
09/20/2018 05:26:17 : [INFO ] CURRENT - PASS    - 1
09/20/2018 05:26:17 : [INFO ] CURRENT - FAIL    - 0
09/20/2018 05:26:17 : [INFO ] CURRENT - ABORTED - 0
09/20/2018 05:26:17 : [INFO ] Checking background cleanup jobs..
09/20/2018 05:26:17 : [INFO ] Removed 0 context files.
09/20/2018 05:26:17 : [INFO ] Cycle Finished.. TC-69895
09/20/2018 05:26:17 : [INFO ]
Test Results Summary
ICA test run on 09/20/2018 04:39:40
ARM Image under test Canonical : UbuntuServer : 18.04-LTS : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:46
XML file: TestConfiguration

VERIFY-SRIOV-FAILSAFE-FOR-DPDK Execution Time: 46.57 minutes
        VERIFY-SRIOV-FAILSAFE-FOR-DPDK : PASS
                        DPDK-TESTPMD : Initial SRIOV : DPDK 18.08.0 : TxPPS : 11229800 : RxPPS : 11229800
                        DPDK-TESTPMD : Synthetic : DPDK 18.08.0 : TxPPS : 69993 : RxPPS : 69993
                        DPDK-TESTPMD : Re-Enable SRIOV : DPDK 18.08.0 : TxPps : 11456061 : RxPps : 11456061
                        DPDK RxPPS : Difference between Initial and Re-Enabled SRIOV : 11229800 : 11456061 : 2.01 %


Logs can be found at .\TestResults\2018-19-09-21-39-24-4693


09/20/2018 05:26:17 : [INFO ] Creating 'Azure-69895-buildlogs.zip' from '.\TestResults\2018-19-09-21-39-24-4693'
09/20/2018 05:26:18 : [INFO ] C:\Users\v-sirebb\Documents\GitHub\LISAv2\Azure-69895-buildlogs.zip created successfully.
09/20/2018 05:26:18 : [INFO ] Copied : .\report\report_TC-69895.xml --> .\report\report_TC-69895-junit.xml
09/20/2018 05:26:18 : [INFO ] Analysing results..
09/20/2018 05:26:18 : [INFO ] PASS  : 1
09/20/2018 05:26:18 : [INFO ] FAIL  : 0
09/20/2018 05:26:18 : [INFO ] ABORT : 0
09/20/2018 05:26:18 : [INFO ] LISAv2 exits with code: 0
```

juhlee-microsoft commented:
@rmsivakanth Please add change description and new parameter information details.

rmsivakanth commented:
Execution is verified on SLES 15 and CentOS 7.5.
```
SLES 15:
09/20/2018 07:43:22 : [INFO ] CleanUP Successful for ICA-RG-DPDK-SRS15-AQCK-636729982292..
09/20/2018 07:43:22 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-SRIOV-FAILSAFE-FOR-DPDK~~~~~~~~~~
09/20/2018 07:43:22 : [INFO ] CURRENT - PASS    - 1
09/20/2018 07:43:22 : [INFO ] CURRENT - FAIL    - 0
09/20/2018 07:43:22 : [INFO ] CURRENT - ABORTED - 0
09/20/2018 07:43:22 : [INFO ] Checking background cleanup jobs..
09/20/2018 07:43:22 : [INFO ] Removed 0 context files.
09/20/2018 07:43:22 : [INFO ] Cycle Finished.. TC-39033
09/20/2018 07:43:22 : [INFO ]
Test Results Summary
ICA test run on 09/20/2018 06:56:58
ARM Image under test SUSE : SLES : 15 : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:46
XML file: TestConfiguration

VERIFY-SRIOV-FAILSAFE-FOR-DPDK Execution Time: 46.33 minutes
        VERIFY-SRIOV-FAILSAFE-FOR-DPDK : PASS
                        DPDK-TESTPMD : Initial SRIOV : DPDK 18.08.0 : TxPPS : 11960223 : RxPPS : 11960223
                        DPDK-TESTPMD : Synthetic : DPDK 18.08.0 : TxPPS : 137758 : RxPPS : 137758
                        DPDK-TESTPMD : Re-Enable SRIOV : DPDK 18.08.0 : TxPps : 11695023 : RxPps : 11695023
                        DPDK RxPPS : Difference between Initial and Re-Enabled SRIOV : 11960223 : 11695023 : 2.22 %


Logs can be found at .\TestResults\2018-19-09-23-56-39-8948


09/20/2018 07:43:23 : [INFO ] Creating 'Azure-39033-buildlogs.zip' from '.\TestResults\2018-19-09-23-56-39-8948'
09/20/2018 07:43:23 : [INFO ] C:\Users\v-sirebb\Documents\GitHub\LISAv2\Azure-39033-buildlogs.zip created successfully.
09/20/2018 07:43:23 : [INFO ] Copied : .\report\report_TC-39033.xml --> .\report\report_TC-39033-junit.xml
09/20/2018 07:43:23 : [INFO ] Analysing results..
09/20/2018 07:43:23 : [INFO ] PASS  : 1
09/20/2018 07:43:23 : [INFO ] FAIL  : 0
09/20/2018 07:43:23 : [INFO ] ABORT : 0
09/20/2018 07:43:23 : [INFO ] LISAv2 exits with code: 0

CentOS 7.5:
09/20/2018 07:47:33 : [INFO ] CleanUP Successful for ICA-RG-DPDK-SRC75-JDAW-636729982397..
09/20/2018 07:47:33 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-SRIOV-FAILSAFE-FOR-DPDK~~~~~~~~~~
09/20/2018 07:47:33 : [INFO ] CURRENT - PASS    - 1
09/20/2018 07:47:33 : [INFO ] CURRENT - FAIL    - 0
09/20/2018 07:47:33 : [INFO ] CURRENT - ABORTED - 0
09/20/2018 07:47:34 : [INFO ] Checking background cleanup jobs..
09/20/2018 07:47:34 : [INFO ] Removed 0 context files.
09/20/2018 07:47:34 : [INFO ] Cycle Finished.. TC-49775
09/20/2018 07:47:34 : [INFO ]
Test Results Summary
ICA test run on 09/20/2018 06:57:09
ARM Image under test Openlogic : CentOS : 7.5 : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:50
XML file: TestConfiguration

VERIFY-SRIOV-FAILSAFE-FOR-DPDK Execution Time: 50.35 minutes
        VERIFY-SRIOV-FAILSAFE-FOR-DPDK : PASS
                        DPDK-TESTPMD : Initial SRIOV : DPDK 18.08.0 : TxPPS : 13219071 : RxPPS : 13219071
                        DPDK-TESTPMD : Synthetic : DPDK 18.08.0 : TxPPS : 78183 : RxPPS : 78183
                        DPDK-TESTPMD : Re-Enable SRIOV : DPDK 18.08.0 : TxPps : 12885617 : RxPps : 12885617
                        DPDK RxPPS : Difference between Initial and Re-Enabled SRIOV : 13219071 : 12885617 : 2.52 %


Logs can be found at .\TestResults\2018-19-09-23-56-52-5942


09/20/2018 07:47:34 : [INFO ] Creating 'Azure-49775-buildlogs.zip' from '.\TestResults\2018-19-09-23-56-52-5942'
09/20/2018 07:47:34 : [INFO ] C:\Users\v-sirebb\Documents\GitHub\LISAv2\Azure-49775-buildlogs.zip created successfully.
09/20/2018 07:47:34 : [INFO ] Copied : .\report\report_TC-49775.xml --> .\report\report_TC-49775-junit.xml
09/20/2018 07:47:34 : [INFO ] Analysing results..
09/20/2018 07:47:34 : [INFO ] PASS  : 1
09/20/2018 07:47:34 : [INFO ] FAIL  : 0
09/20/2018 07:47:35 : [INFO ] ABORT : 0
09/20/2018 07:47:35 : [INFO ] LISAv2 exits with code: 0
```

rmsivakanth commented on Testscripts/Windows/VERIFY-SRIOV-FAILSAFE-FOR-DPDK.ps1:
```diff
@@ -0,0 +1,222 @@
+                $clientVMData = $vmData
+                $noClient = $false
+            }
+            elseif ($vmData.RoleName -imatch "server") {
```
add else block here with error message.

rmsivakanth commented:
Execution also verified after addressing all comments and minor fix.
```
09/20/2018 20:24:48 : [INFO ] ~~~~~~~~~~~~~~~TEST END : VERIFY-SRIOV-FAILSAFE-FOR-DPDK~~~~~~~~~~
09/20/2018 20:24:48 : [INFO ] CURRENT - PASS    - 1
09/20/2018 20:24:48 : [INFO ] CURRENT - FAIL    - 0
09/20/2018 20:24:48 : [INFO ] CURRENT - ABORTED - 0
09/20/2018 20:24:48 : [INFO ] Checking background cleanup jobs..
09/20/2018 20:24:48 : [INFO ] Removed 0 context files.
09/20/2018 20:24:48 : [INFO ] Cycle Finished.. TC-11601
09/20/2018 20:24:48 : [INFO ]
Test Results Summary
ICA test run on 09/20/2018 19:34:21
ARM Image under test Openlogic : CentOS : 7.5 : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:50
XML file: TestConfiguration

VERIFY-SRIOV-FAILSAFE-FOR-DPDK Execution Time: 50.4 minutes
        VERIFY-SRIOV-FAILSAFE-FOR-DPDK : PASS
                        DPDK-TESTPMD : Initial SRIOV : DPDK 18.08.0 : TxPPS : 11799861 : RxPPS : 11785182
                        DPDK-TESTPMD : Synthetic : DPDK 18.08.0 : TxPPS : 306492 : RxPPS : 100042
                        DPDK-TESTPMD : Re-Enable SRIOV : DPDK 18.08.0 : TxPps : 11440791 : RxPps : 11377884
                        DPDK RxPPS : Difference between Initial and Re-Enabled SRIOV : 11785182 : 11377884 : 3.46 %


Logs can be found at .\TestResults\2018-20-09-12-34-02-2983


09/20/2018 20:24:48 : [INFO ] Creating 'Azure-11601-buildlogs.zip' from '.\TestResults\2018-20-09-12-34-02-2983'
09/20/2018 20:24:49 : [INFO ] C:\Users\v-sirebb\Documents\GitHub\LISAv2\Azure-11601-buildlogs.zip created successfully.
09/20/2018 20:24:49 : [INFO ] Copied : .\report\report_TC-11601.xml --> .\report\report_TC-11601-junit.xml
09/20/2018 20:24:49 : [INFO ] Analysing results..
09/20/2018 20:24:49 : [INFO ] PASS  : 1
09/20/2018 20:24:49 : [INFO ] FAIL  : 0
09/20/2018 20:24:49 : [INFO ] ABORT : 0
09/20/2018 20:24:49 : [INFO ] LISAv2 exits with code: 0
```
